### PR TITLE
fix: node_modules sourceMaps

### PR DESCRIPTION
**Pull request by m1ga**
Currently if you have npm modules that have source map files compiling will fail with:
```
[ERROR] TypeError [ERR_INVALID_ARG_TYPE]: The "paths[1]" argument must be of type string. Received undefined
    at Object.resolve (node:path:1101:7)
    at Object.analyzeJs (/home/miga/.titanium/mobilesdk/linux/12.5.0/node_modules/node-titanium-sdk/lib/jsanalyze.js:201:26)
    at ProcessJsTask.transformAndCopy (/home/miga/.titanium/mobilesdk/linux/12.5.0/cli/lib/tasks/process-js-task.js:308:31)
    at AndroidBuilder.<anonymous> (/home/miga/.titanium/mobilesdk/linux/12.5.0/cli/lib/tasks/process-js-task.js:218:17)
    at data.result (file:///usr/local/lib/node_modules/titanium/src/cli.js:322:15)
    at new Promise (<anonymous>)
    at file:///usr/local/lib/node_modules/titanium/src/cli.js:320:26
```

The fix is to exclude files that are in the node_modules folder.

**Test**
* create a new app `ti create --alloy`
* create a `app/lib` folder
* go into that folder and install e.g. underscore using npm
* compile the app

### CODE REVIEW

File: lib/jsanalyze.js [hunk 1]
```diff
@@ -168,8 +168,10 @@ exports.analyzeJs = function analyzeJs(contents, opts = {}) {
 	}
 
 	let sourceFileName = opts.filename; // used to point to correct original source in sources/sourceRoot of final source map
+	let sourceMap = (sourceFileName && sourceFileName.includes('/node_modules/')) ? false : opts.sourceMap;
```
Comment (cb1kenobi) [inline lib/jsanalyze.js:171]: **suggestion (non-blocking):** Super nitpick‏

```suggestion
	const sourceMap = (sourceFileName && sourceFileName.includes('/node_modules/')) ? false : opts.sourceMap;
```

**Review by cb1kenobi — APPROVED** [review]
Looks good to me!

**Review by cb1kenobi — APPROVED** [review]

**Review by hansemannn — APPROVED** [review]

**Comment by hansemannn** [comment]
We still need to bump the version and release to npm I think?

**Comment by m1ga** [comment]
correct. Version: I wasn't sure if there is a workflow for that or if I need to do it by hand

**Pull request merged**